MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult+Internal.swift

```diff
@@ -26,25 +26,83 @@ import Foundation
+        params.forceRefresh = forceRefresh
+        params.correlationId = correlationId
+
+        let challengeTypes = MSALNativeAuthPublicClientApplication.convertChallengeTypes(configuration.challengeTypes)
+        let authority = try? MSALCIAMAuthority(url: configuration.authority.url)
+        let config = MSALPublicClientApplicationConfig(clientId: configuration.clientId,
+                                                       redirectUri: configuration.redirectUri,
+                                                       authority: authority)
+        config.bypassRedirectURIValidation = configuration.redirectUri == nil
+
+        guard let client = try? MSALNativeAuthPublicClientApplication(configuration: config, challengeTypes: challengeTypes)
+        else {
+            MSALLogger.log(
+                            level: .error,
+                            context: context,
+                            format: "Config or challenge types unexpectedly found nil."
+                        )
+            Task { await delegate.onAccessTokenRetrieveError(
+                error: RetrieveAccessTokenError(type: .generalError,
+                                                correlationId: correlationId ?? context.correlationId())) }
+            return
+        }
+
+        client.acquireTokenSilent(with: params) { [weak self] result, error in
+            guard let self = self else { return }
+
+            if let error = error as? NSError {
+                let accessTokenError = self.createRetrieveAccessTokenError(error: error, context: context)
+                Task { await delegate.onAccessTokenRetrieveError(error: accessTokenError) }
+                return
+            }
+
+            if let result = result {
+                let delegateDispatcher = CredentialsDelegateDispatcher(delegate: delegate, telemetryUpdate: nil)
+                let accessTokenResult = MSALNativeAuthTokenResult(accessToken: result.accessToken,
+                                                                  scopes: result.scopes,
+                                                                  expiresOn: result.expiresOn)
+                Task { await delegateDispatcher.dispatchAccessTokenRetrieveCompleted(result: accessTokenResult, correlationId: result.correlationId) }
+                return
+            }
+
+            Task { await delegate.onAccessTokenRetrieveError(error: RetrieveAccessTokenError(type: .generalError,
+                                                                                             correlationId: correlationId ?? context.correlationId())) }
         }
     }
 
-    func refreshData(authTokens: MSALNativeAuthTokens, account: MSALAccount) {
-        self.authTokens = authTokens
-        self.account = account
+    func createRetrieveAccessTokenError(error: NSError, context: MSALNativeAuthRequestContext) -> RetrieveAccessTokenError {
```

Comment:
Coming from this [comment](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2159#discussion_r1621194657), I have a couple questions about this function. It does three attempts to get the error:

1. Get the error from error.userInfo[NSUnderlyingErrorKey] + message + correlationId.
2. If any of the 1) fails, it tries to get the message + code + correlationId.
3. If any of the 1) fails, it creates a generic error.

Here are the questions:

- Why the code is not retrieved in 1)?
- Why the code in ⁠2) differs from the code in 3)? (I think this is the same as this [comment](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2159/files#r1621077026))
    - In 2) is error.userInfo[MSALInternalErrorCodeKey]
    - In 3) is error.code
- In both 1) and 2) we still can get an empty correlationId (the function `correlationIdFromMSALError()` can return an empty String. Shouldn't we pass the context to that function and return the `context.correlationId` in case the one from the api error is not available?

Reply:
Me and Diego discussed this comment in a call. What we clarified:
- the code isn't retrieved in 1) because `MSALInternalErrorCodeKey` contains only error codes generated internally by the SDK. We should return only error codes related to eSTS
- We should not parse the error in this case. Code updated
- `correlationIdFromMSALError()` function can return an UUID nil (UUID of an empty string is nil)